src/typechecker.erl

```diff
@@ -1442,7 +1461,8 @@ merge_types(Tys) ->
 add_types_pats([], [], _TEnv, VEnv) ->
     VEnv;
 add_types_pats([Pat | Pats], [Ty | Tys], TEnv, VEnv) ->
-    add_types_pats(Pats, Tys, TEnv, add_type_pat(Pat, Ty, TEnv, VEnv)).
+    NormTy = normalize(Ty, TEnv),
+    add_types_pats(Pats, Tys, TEnv, add_type_pat(Pat, NormTy, TEnv, VEnv)).
```

Comment:
What about the normalization here? Do we need to restore the original type here too?

Reply:
ops, forgot about this place, fixed in a new commit